code/modules/research/designs/weapon_designs.dm

```diff
@@ -23,13 +23,14 @@
 	category = list("Firing Pins")
 
 /datum/design/stunrevolver
-	name = "Stun Revolver"
+	name = "Shock Revolver"
 	desc = "A high-tech revolver that fires internal, reusable taser cartridges in a revolving cylinder. The cartridges can be recharged using conventional rechargers."
 	id = "stunrevolver"
-	req_tech = list("combat" = 3, "materials" = 3, "powerstorage" = 2)
+	req_tech = list("combat" = 3, "materials" = 3, "powerstorage" = 5)
 	build_type = PROTOLATHE
-	materials = list(MAT_METAL = 4000, MAT_GLASS = 1000)
-	build_path = /obj/item/weapon/gun/energy/stunrevolver
+	materials = list(MAT_METAL = 10000, MAT_GLASS = 10000, MAT_SILVER = 10000)
```

Comment:
Good Lord, Kor. These prices are extreme mech level. Please do not tell me that you plan to balance everything in such a restrictive manner; you will have it such that Science can hardly make anything! Broke people tend not to want to share, and it would be a real shame if it became even more difficult to get my staff to give people things. 

Please be very mindful of how you price things, especially after you gutted efficiency upgrades.